utils/src/test/kotlin/UtilsTest.kt

```diff
@@ -58,6 +58,83 @@ class UtilsTest : WordSpec({
         }
     }
 
+    "filterVersionNames" should {
+        "return an empty list for a blank version" {
+            val names = listOf("dummy")
+
+            filterVersionNames("", names) shouldBe emptyList<String>()
+            filterVersionNames(" ", names) shouldBe emptyList<String>()
+        }
+
+        "find names separated by underscores" {
+            val names = listOf(
+                    "A02",
+                    "A03",
+                    "A04",
+                    "DEV0_9_3",
+                    "DEV0_9_4",
+                    "DEV_0_9_7",
+                    "Exoffice",
+                    "PROD_1_0_1",
+                    "PROD_1_0_2",
+                    "PROD_1_0_3",
+                    "before_SourceForge",
+                    "before_debug_changes"
+            )
+
+            filterVersionNames("1.0.3", names).joinToString("\n") shouldBe "PROD_1_0_3"
+            filterVersionNames("1", names).joinToString("\n") shouldBe ""
+        }
+
+        "find names separated by dots" {
+            val names = listOf(
+                    "0.10.0",
+                    "0.10.1",
+                    "0.5.0",
+                    "0.6.0",
+                    "0.7.0",
+                    "0.8.0",
+                    "0.9.0"
+            )
+
+            filterVersionNames("0.10.0", names).joinToString("\n") shouldBe "0.10.0"
+            filterVersionNames("1", names).joinToString("\n") shouldBe ""
```

Comment:
Also test that "0.10" and "10.0" do not return results.

Reply:
Done.